src/server/serve_static.cpp

```
@@ -297,28 +297,34 @@ operator()(route_params& rp) const
+                co_return route_error(ec2);
+            continue;
+        }
+
         auto const to_read = static_cast<std::size_t>(
-            (std::min)(remaining, static_cast<std::int64_t>(buf_size)));
+            (std::min)(remaining,
+                static_cast<std::int64_t>(bufs[0].size())));
 
-        auto const n1 = f.read(buffer, to_read, ec);
+        auto const n1 = f.read(bufs[0].data(), to_read, ec);
         if(ec.failed() || n1 == 0)
```

Cppcheck (2.19.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)